internal/codespaces/rpc/invoker.go

```diff
@@ -10,48 +10,64 @@ import (
 	"strconv"
 	"time"
 
-	"github.com/cli/cli/v2/internal/codespaces/grpc/jupyter"
+	"github.com/cli/cli/v2/internal/codespaces/rpc/jupyter"
 	"github.com/cli/cli/v2/pkg/liveshare"
-	"golang.org/x/crypto/ssh"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials/insecure"
 	"google.golang.org/grpc/metadata"
 )
 
 const (
 	ConnectionTimeout = 5 * time.Second
-	RequestTimeout    = 30 * time.Second
+	requestTimeout    = 30 * time.Second
 )
 
 const (
 	codespacesInternalPort        = 16634
 	codespacesInternalSessionName = "CodespacesInternal"
 )
 
-type Client struct {
+type Invoker interface {
+	Close() error
+	StartJupyterServer(ctx context.Context) (int, string, error)
+	RebuildContainer(ctx context.Context, full bool) error
+	StartSSHServer(ctx context.Context) (int, string, error)
+	StartSSHServerWithOptions(ctx context.Context, options liveshare.StartSSHServerOptions) (int, string, error)
+}
+
+type invoker struct {
 	conn          *grpc.ClientConn
 	token         string
+	session       liveshare.LiveshareSession
 	listener      net.Listener
 	jupyterClient jupyter.JupyterServerHostClient
 	cancelPF      context.CancelFunc
 }
 
-type liveshareSession interface {
-	KeepAlive(string)
-	OpenStreamingChannel(context.Context, liveshare.ChannelID) (ssh.Channel, error)
-	StartSharing(context.Context, string, int) (liveshare.ChannelID, error)
+// Connects to the internal RPC server and returns a new invoker for it
+func CreateInvoker(ctx context.Context, session liveshare.LiveshareSession, token string) (Invoker, error) {
```

Comment:
Last question I think, what is `token` used for? It looks like it's always an empty string?

Reply:
It's not really used so I removed it. The initial idea was to have a custom bearer token for additional authentication but since we don't publicly forward the user's gRPC port, there's not really a need for it. The server still checks for the auth header but doesn't actually check the value.